libgnucash/engine/gnc-numeric.cpp

```diff
@@ -208,17 +208,20 @@ numeric_from_scientific_match(smatch &m)
 }
 
 GncNumeric::GncNumeric(const std::string &str, bool autoround) {
+    static const std::string integer("(-?[0-9]+)");
     static const std::string numer_frag("(-?[0-9]*)");
     static const std::string denom_frag("([0-9]+)");
```

Comment:
`numer_frag` and `denom_frag` need new names because you're using `denom_frag` as a numerator.
How about `signed_int`, `opt_signed_int` and `unsigned_int`?